packages/driver/src/cypress/cy.js

```diff
@@ -673,13 +677,50 @@ const create = function (specWindow, Cypress, Cookies, state, config, log) {
     return state('index', queue.length)
   }
 
-  const fail = function (err, runnable) {
+  const getInvocationStack = (err) => {
+    const current = state('current')
+    const currentName = current && current.get('name')
+
+    if (err.stack && (
+      // cy.then will not error on its own, but will have an
+      // invocation stack, so we need to prefer the actual error's stack
+      (currentName === 'then') ||
```

Comment:
rather than anchor ourselves to `then` we should come up with a pattern to detect when callback functions are invoked - the same should apply for `cy.each()`

TODO: create a custom command which takes a callback and verify its stack trace is correct